client/engine/engine.go

```diff
@@ -548,6 +607,48 @@ func (e *Engine) attemptProgress(objective protocols.Objective) (outgoing Engine
 	return
 }
 
+// handleNotifications takes an objective and constructs notifications for any related channels for that objective.
+func (e *Engine) handleNotifications(o protocols.Objective) (EngineEvent, error) {
```

Comment:
Super nit. I like when `handleFoo` takes an argument of type `foo`. Could this be renamed to e.g. `generateNotifications`?

Reply:
7db1dbd649fd23863b554b34576c2b9f7f793835